crates/store/src/db/models/queries.rs

```diff
@@ -1072,7 +1071,155 @@ pub(crate) fn unconsumed_network_notes(
+    // ORDER BY rowid
+    // LIMIT ?
+    #[allow(
+        clippy::items_after_statements,
+        reason = "It's only relevant for a single call function"
+    )]
+    type RawLoadedTuple = (
+        i64,             // block_num
+        i32,             // batch_index
+        i32,             // note_index
+        Vec<u8>,         // note_id
+        i32,             // note_type
+        Vec<u8>,         // sender
+        i32,             // tag
+        i64,             // aux
+        i64,             // execution_hint
+        Vec<u8>,         // merkle_path
+        Option<Vec<u8>>, // assets
+        Option<Vec<u8>>, // inputs
+        Option<Vec<u8>>, // script
+        Option<Vec<u8>>, // serial_num
+        i64,             // rowid (from sql::<BigInt>("notes.rowid"))
+    );
+
+    #[allow(
+        clippy::items_after_statements,
+        reason = "It's only relevant for a single call function"
+    )]
+    fn split_into_raw_note_record_and_implicit_row_id(
+        tuple: RawLoadedTuple,
+    ) -> (NoteRecordRaw, i64) {
+        (
+            NoteRecordRaw {
+                block_num: tuple.0,
+                batch_index: tuple.1,
+                note_index: tuple.2,
+                note_id: tuple.3,
+                note_type: tuple.4,
+                sender: tuple.5,
+                tag: tuple.6,
+                aux: tuple.7,
+                execution_hint: tuple.8,
+                merkle_path: tuple.9,
+                assets: tuple.10,
+                inputs: tuple.11,
+                script: tuple.12,
+                serial_num: tuple.13,
+            },
+            tuple.14,
+        )
+    }
+
+    let raw = SelectDsl::select(
+        schema::notes::table.left_join(
+            schema::note_scripts::table
+                .on(schema::notes::script_root.eq(schema::note_scripts::script_root.nullable())),
+        ),
+        (
+            schema::notes::block_num,
+            schema::notes::batch_index,
+            schema::notes::note_index,
+            schema::notes::note_id,
+            // metadata
+            schema::notes::note_type,
+            schema::notes::sender,
+            schema::notes::tag,
+            schema::notes::aux,
+            schema::notes::execution_hint,
+            schema::notes::inclusion_path,
+            // details
+            schema::notes::assets,
+            schema::notes::inputs,
+            schema::note_scripts::script.nullable(),
+            schema::notes::serial_num,
+            rowid_sel.clone(),
+        ),
+    )
+    .filter(schema::notes::execution_mode.eq(0_i32))
+    .filter(schema::notes::tag.eq(tag_to_raw_sql(u32::from(network_account_id_prefix).into())))
+    .filter(schema::notes::block_num.le(block_number_to_raw_sql(latest_block_num)))
+    .filter(
+        schema::notes::consumed_block_num
+            .is_null()
+            .or(schema::notes::consumed_block_num.gt(block_number_to_raw_sql(latest_block_num))),
```

Comment:
We are including all blocks that fullfil the following conditions:
`x.ececuttion_mode == Local`
`x.block_num  <= latest_block_num`
`x.tag == tag` 
`consumed_block_num == NIL || x.consumed_block_num > latest_block_num` <-- could you elaborate? To my understanding this means not consumed yet or will be consumed in a block in the future (relative to `latest_block_num`)?

Reply:
> `consumed_block_num == NIL || x.consumed_block_num > latest_block_num` <-- could you elaborate?

Not consumed or consumed after the threshold specified by the query. You can think of that block number specifying the snapshot of state that scopes the query.